pkg/controller/pxc/controller.go

```diff
@@ -364,6 +355,29 @@ func (r *ReconcilePerconaXtraDBCluster) deploy(cr *api.PerconaXtraDBCluster) err
 	return nil
 }
 
+func (r *ReconcilePerconaXtraDBCluster) reconcileConfigMap(cr *api.PerconaXtraDBCluster) error {
+	if cr.Spec.PXC.Configuration != "" {
+		stsApp := statefulset.NewNode(cr)
+		ls := stsApp.Labels()
+		configMap := configmap.NewConfigMap(cr, ls["app.kubernetes.io/instance"]+"-"+ls["app.kubernetes.io/component"])
+		err := setControllerReference(cr, configMap, r.scheme)
+		if err != nil {
+			return err
+		}
+		err = r.client.Create(context.TODO(), configMap)
+		if err != nil && !errors.IsAlreadyExists(err) {
+			return fmt.Errorf("create ConfigMap: %v", err)
+		} else {
+			err = r.client.Update(context.TODO(), configMap)
+			if err != nil {
+				return fmt.Errorf("update ConfigMap: %v", err)
+			}
+		}
```

Comment:
Successful create will be followed by update (`err == nil` fall into `else` condition).
I sugest:
```go
		err = r.client.Create(context.TODO(), configMap)
		if err != nil && errors.IsAlreadyExists(err) {
			err = r.client.Update(context.TODO(), configMap)
			if err != nil {
				return fmt.Errorf("update ConfigMap: %v", err)
			}
		} else if err != nil {
			return fmt.Errorf("create ConfigMap: %v", err)
		}
```

Reply:
Thanks